pkg/pushsync/pushsync.go

```diff
@@ -493,14 +476,46 @@ func (ps *PushSync) pushPeer(ctx context.Context, peer swarm.Address, ch swarm.C
 		return nil, true, fmt.Errorf("invalid receipt. chunk %s, peer %s", ch.Address(), peer)
 	}
 
-	err = ps.accounting.Credit(peer, receiptPrice, originated)
+	err = ps.accounting.Credit(peer, receiptPrice, origin)
 	if err != nil {
 		return nil, true, err
 	}
 
 	return &receipt, true, nil
 }
 
+func (ps *PushSync) pushToNeighbourhood(ctx context.Context, skiplist []swarm.Address, ch swarm.Chunk, origin bool, originAddr swarm.Address) {
+	count := 0
+	// Push the chunk to some peers in the neighborhood in parallel for replication.
+	// Any errors here should NOT impact the rest of the handler.
+	_ = ps.topologyDriver.EachPeer(func(peer swarm.Address, po uint8) (bool, bool, error) {
```

Comment:
just a suggestion, but using `ClosestPeer` with the `skiplist+origin` as peers to skip could reduce the amount of code here significantly.